static/js/components/UploadSpectrum.jsx

```diff
@@ -0,0 +1,494 @@
+      label: "Comment",
+      options: {
+        filter: false,
+        sort: true,
+      },
+    });
+  }
+
+  const data_columns = [
+    {
+      name: "wavelength",
+      label: "Wavelength (Angstroms)",
+      options: {
+        filter: false,
+        sort: true,
+      },
+    },
+    {
+      name: "flux",
+      label: "Flux",
+      options: {
+        filter: false,
+        sort: true,
+      },
+    },
+  ];
+
+  if (parsed && "errors" in parsed) {
+    data_columns.push({
+      name: "error",
+      label: "Error",
+      options: {
+        filter: false,
+        sort: true,
+      },
+    });
+  }
+
+  const uploadFormSchema = {
+    type: "object",
+    properties: {
+      file: {
+        type: "string",
+        format: "data-url",
+        title: "Spectrum file",
+      },
+      group_ids: {
+        type: "array",
+        title: "Share with...",
+        items: {
+          type: "integer",
+          anyOf: groups.map((group) => ({
+            enum: [group.id],
+            title: group.name,
+          })),
+        },
+        uniqueItems: true,
+      },
+      mjd: {
+        type: "number",
+        title: "Observation MJD",
+      },
+      instrument_id: {
+        type: "integer",
+        title: "Instrument",
+        enum: instruments.map((instrument) => instrument.id),
+        enumNames: instruments.map((instrument) => {
+          const telescope = telescopes.find(
+            (t) => t.id === instrument.telescope_id
+          );
+          let name = "";
+          if (telescope) {
+            name += `${telescope.nickname} / `;
+          }
+          name += instrument.name;
+          return name;
+        }),
+      },
+      wave_column: {
+        type: "integer",
+        default: 0,
+        title: "0-Based Wavelength Column Index",
+      },
+      flux_column: {
+        type: "integer",
+        default: 1,
+        title: "0-Based Flux Column Index",
+      },
+      has_fluxerr: {
+        type: "string",
+        default: "No",
+        title: "Does your spectrum have flux errors?",
+        enum: ["No", "Yes"],
+      },
+    },
+    required: [
+      "has_fluxerr",
+      "mjd",
+      "wave_column",
+      "flux_column",
+      "instrument_id",
+    ],
+    dependencies: {
+      has_fluxerr: {
+        oneOf: [
+          {
+            properties: {
+              has_fluxerr: {
+                enum: ["No"],
+              },
+            },
+          },
+          {
+            properties: {
+              has_fluxerr: {
+                enum: ["Yes"],
+              },
+              fluxerr_column: {
+                type: "integer",
+                default: null,
+                title: "0-Based Flux Error Column Index",
+              },
+            },
+            required: ["has_fluxerr", "fluxerr_column"],
+          },
+        ],
+      },
+    },
+  };
+
+  const parseAscii = ({ formData }) => {
+    dispatch({ type: Actions.RESET_PARSED_SPECTRUM });
```

Comment:
No need. The `parsed` state will be overwritten by the action response whether it is null or not when the response arrives or not. This is done just to  clear the page so that the user is not looking at outdated components.  No need to wait for it.